shortcut/base.py

```diff
@@ -15,77 +17,162 @@ class ShortCutter(object):
             file path `/path/to/my_program` or a simple application name 
             `my_program`.
-
         :param str shortcut_directory:
             The directory path where the shortcut should be created.
+        :param str target_name:
+            Name of the shortcut without extension (.lnk would be appended if needed).
+        :param bool virtual:
+            whether to allow shortcuts to yet non-existing files
 
         Returns a tuple of (target_name, target_path, shortcut_file_path)
         """
-        # get the target name by getting the file name and removing the extension
-        target_name = os.path.splitext(os.path.basename(target))[0]
+        if target_name is None:
+            # get the target name by getting the file name and removing the extension
+            target_name = os.path.splitext(os.path.basename(target))[0]
 
         # find for the target path  
         target_path = self.find_target(target)
 
-        shortcut_file_path = self._create_shortcut_file(target_name, target_path, shortcut_directory)
+        # Create temporal file in order to create shortcut in virtual mode:
+        clean = False
+        if virtual and (target_path is None):
+            try:
+                if not os.path.isdir(os.path.dirname(target)):
+                    os.makedirs(os.path.dirname(target))
+                open(target, 'a').close()
+                target_path = self.find_target(target)
+                clean = True
+            except (OSError, IOError):
+                self.err_file.write(''.join(traceback.format_exc()))
+
+        # Create shortcut to the target_path:
+        # noinspection PyBroadException
+        try:
+            shortcut_file_path = self._create_shortcut_file(target_name, target_path, shortcut_directory)
+        except:
+            shortcut_file_path = None
+            self.err_file.write(''.join(traceback.format_exc()))
+
+        # Delete temporal file:
+        if clean:
+            os.remove(target)
+
+        return target_name, target_path, shortcut_file_path
+
+    def create_shortcut_to_dir(self, target_path, shortcut_directory, target_name=None):
+        """
+        Creates a shortcut to a direcrory.
 
-        return (target_name, target_path, shortcut_file_path)
+        :param str target_path:
+            The target directory path to create a shortcut for.
+        :param str shortcut_directory:
+            The directory path where the shortcut should be created.
+        :param str target_name:
+            Name of the shortcut without extension (.lnk would be appended if needed).
 
-    #needs overriding
+        Returns a tuple of (target_name, target_path, shortcut_file_path)
+        """
+        if target_name is None:
+            # get the target_name by getting the target dir name
+            target_name = os.path.basename(target_path)
+
+        # Create target_path if it doesn't exist:
+        if not os.path.isdir(target_path):
```

Comment:
See above, I would expect an error to be raised if the directory didnt exist unless the virtual param had been set.

Reply:
I agree. I guess consistent behavior would be better. I'll add `virtual` kwarg support to shortcut to dir.